pipelines/common/src/main/java/com/google/fhir/analytics/view/ViewSchema.java

```diff
@@ -160,4 +170,133 @@ public static void setValueInStatement(
       }
     }
   }
+
+  /**
+   * Loads flat view values into GenericRecords by adding each RowElement value to its respective
+   * column in a GenericRecord. Each GenericRecord contains values for one FlatRow
+   *
+   * @param rows RowList containing materialized view data
+   * @param vDef The current ViewDefinition used to define the flat view
+   * @return List of populated GenericRecords
+   */
+  public static List<GenericRecord> setValueInRecord(RowList rows, ViewDefinition vDef) {
+    List<GenericRecord> recordList = new ArrayList<>();
+
+    // Add each RowElement value to its respective column in the GenericRecord
+    Schema flatSchema = ViewSchema.getAvroSchema(vDef);
+    for (FlatRow flatRow : rows.getRows()) {
+      GenericRecord currentRecord = new GenericData.Record(flatSchema);
+      for (RowElement e : flatRow.getElements()) {
+        if (e.getPrimitive() != null) {
+          if (ViewApplicator.ID_TYPE.equals(e.getColumnInfo().getInferredType())) {
+            currentRecord.put(e.getColumnInfo().getName(), e.getString());
+          } else {
+            String elementType =
+                e.getColumnInfo().getType() == null ? "any" : e.getColumnInfo().getType();
+            switch (elementType) {
+              case "boolean":
+                currentRecord.put(e.getColumnInfo().getName(), e.<Boolean>getPrimitive());
+                break;
+              case "integer":
+              case "unsignedInt":
+              case "integer64":
+              case "decimal":
+                currentRecord.put(e.getColumnInfo().getName(), e.getPrimitive());
+                break;
+              case "date":
+              case "dateTime":
+              case "instant":
+              case "time":
+              case "base64Binary":
+              case "canonical":
+              case "code":
+              case "id":
+              case "markdown":
+              case "oid":
+              case "string":
+              case "uri":
+              case "url":
+              case "uuid":
+              default:
+                currentRecord.put(e.getColumnInfo().getName(), e.getString());
+                break;
+            }
+          }
+        } else {
+          // Currently arrays are not supported; also type inference is not enabled.
```

Comment:
We should be able to support arrays in Avro records; why are we not supporting that?

Reply:
Good catch, now supporting collections as string arrays.